packages/discussions/lib/discussionPreferences.js

```diff
@@ -80,6 +81,92 @@ module.exports = async ({
   if (existingDP) {
     await transaction.public.discussionPreferences.updateOne(findQuery, updateQuery, options)
   } else {
-    await transaction.public.discussionPreferences.insert(updateQuery, options)
+    await Promise.all([
+      transaction.public.discussionPreferences.insert(updateQuery, options),
+      loaders.Discussion.Commenter.discussionPreferences.clear({
+        userId: userId,
+        discussionId: discussion.id
+      })
```

Comment:
What the reason that this only needs to happen on `insert` and not after `updateOne`? IMO could be moved after the `if (existingDP)` block.

Reply:
Argh, this is a mistake.

I only wanted to clear the cache on `update` (and I actually went for `insert`), but of course, an empty response would be cached in the loader too.